git-workflow.md

```diff
@@ -0,0 +1,101 @@
+
+### Acceptance
+The development branch is merged into `develop`, which should trigger a deployment
+on CircleCI.
+
+```
+master      ----*
+                 \
+feature/#n        \-*-*-*-*
+                           \
+develop     ----------------m
+```
+
+### Production
+Deployments to production are completed by creating a release branch from master and
+successively merging the selected\* feature branches into the release branch. This allows features
+to be 'cherry-picked' into production without having to wait for all features in develop
+to receive Product Owner approval. 
+
+\**Waffle is currently used to to mark issues by adding them to the "Deploying" column. Both relsr and flightplan use the
+waffle labels to identify issues for the next release.*
+
+```
+master          ---------*-------------------m-------
+                   \      \                 /
+release/20171231-101500    \----m-------m--*
+                     \         /       /
+feature/#n            \-*-*-*-*       / 
+                       \       \     / 
+feature/#x              \-*-*-*-\-*-* 
+                         \       \   \ 
+bug/#b                    \-*-*   \   \ 
+                               \   \   \ 
+develop     --------------------m---m---m----------
+```
+
+To create the release branch and PR you can either use [relsr](https://github.com/jcleary/relsr) on the command-line or allow [flightplan](https://flightplan.createl.io) to create
+them for you on the deployment schedule (currently PR at 10:15, merge at 10:30 daily).
+
+### Resolving Conflicts
+If GutHub warns you of conflicts in your PR, resolve the conflicts by merging your branch into develop on the command line. 
+Do you use the GutHub tools or the command line to merge develop into your branch as this will create deployment dependencies. 
+The process looks like this:
+```bash
+git checkout develop
+git pull
+git merge feature/#1234-strong-passwords
+# resolve conflicts
+git commit
+git push
+```
+
+When you push develop, GitHub will close the PR for you automatically. 
+
+### Rework
+1. Pull the original branch
+1. Create a rework branch without the # (e.g rework/1234-fix-breaking-test-part-1)
+1. When ready, open the PR back to the original branch - this will ensure the PR is just for the most recent changes.
+1. After the PR is merged, manually merge the feature branch back into `develop` to get your changes on to Staging.
+1. If there is subsequent rework, repeat the above steps, creating a new rework branch.
+
+### Special Workflow (Grouping Issues)
+From time to time it may become necessary to deploy certain features together. In these cases please ensure you follow the special workflow:
+
+1. Create a issue for combining the features. (e.g. Issue 5678 - Languages and React Group Deployment), referencing the original issues in the description.
+1. Create a branch for the issue ( e.g. `group/#5678-languages-and-react` )
+1. Merge all the feature branches into the group branch.
```

Comment:
What about deploying to staging (i.e. merging to develop or fastlane)?

Reply:
Not sure I follow. So do the individual feature branches get PR-ed to `develop`, or does the entire group branch?

Also, would it be better to create the group branch up front?